src/environment.rs

```diff
@@ -283,6 +283,7 @@ impl<I: Interface + Clone + From<Proxy<I>> + AsRef<Proxy<I>>> GlobalHandler<I> f
         version: u32,
         _: DispatchData,
     ) {
+        let version = I::VERSION.min(version);
```

Comment:
If anything I'd want a sort of compile time assertion or an unconditional panic if you try to bind a version above the API one.

Reply:
You can't do that from here.  The version passed in to this function is the maximum version the server supports; there is no information about the version the client knows about other than the maximum version our XML knows about.

Reply:
Yeah, be aware that it's a 0.16.0, it's a bit different.

Reply:
Yeah I noticed that after the review comment, I'm less sure about 0.16

Reply:
How it's done is the only way to do as far as I remember. The alternative is to explicitly bind the version we support though, not just the one maximum in XML we use.

@danieldg maybe we should just bind the actual version we support and optionally clam with the interface one? It should be more robust in case `I::VERSION` will change.

Reply:
What is "the version we support", and how do you get access to it here in the SimpleGlobal impl?

Note that `I::VERSION` shouldn't change unless the wayland-client repository updates its internal wayland.xml dependency.

Reply:
I just don't remember whether the `wayland.xml` update was a breaking change or not. I think it's not a breaking change as long as enum versions are the same.

You're correct, we can't do anything with `SimpleGlobal` here due to how it's all designed and not do a breaking change.

Reply:
Basically this patch is not a good way to do it if wayland.xml does change (although it would actually fix the error message that clients are hitting, they'll likely hit the  `_ => unreachable!()` branch of an event-handling enum written when the enums had fewer variants.  Thanks to clippy, ignoring unknown variants requires annotations but panicking on them does not.

Reply:
Yeah, but I don't think the wayland.xml of wayland 0.29 will be updated. We just simply don't have other option as of now.